Cython/Utility/Builtins.c

```diff
@@ -303,6 +303,41 @@ static PyObject *__Pyx_PyLong_AbsNeg(PyObject *n) {
 #define __Pyx_PyNumber_Power2(a, b) PyNumber_Power(a, b, Py_None)
 
 
+//////////////////// divmod_int.proto //////////////////
+
+static CYTHON_INLINE PyObject* __Pyx_divmod_int(int a, int b); /*proto*/
+
+
+//////////////////// divmod_int //////////////////
+
+static CYTHON_INLINE PyObject* __Pyx_divmod_int(int a, int b) {
+    PyObject* result_tuple;
+    // Python and C/C++ use different algorithm in calculating quotients and remainders.
+    // This results in different answers between Python and C/C++
+    // when the dividend is negative and the divisor is positive and vice versa.
+    if ((a < 0 && b > 0) || (a > 0 && b < 0)) {
+        // see CMath.c :: DivInt and ModInt utility code
+        int q = a / b;
+        int r = a - q * b;
+        q -= ((r != 0) & ((r ^ b) < 0));
+        r += ((r != 0) & ((r ^ b) < 0)) * b;
+        result_tuple = PyTuple_Pack(2, PyLong_FromLong(q), PyLong_FromLong(r));
+        if (unlikely(!result_tuple))
+            return NULL;
+        return result_tuple;
```

Comment:
Here, the error handling is redundant.
```suggestion
        return result_tuple;
```

Reply:
Removed in the latest commit. Please review it. Thank you.